src/compute-engine/boxed-expression/match.ts

```diff
@@ -352,82 +403,217 @@ function matchArguments(
+            // and capture the optional wildcards subsequently.
+            let nextAppPattern = nextPattern;
+            let nextAppPatternIndex = i + 1;
+
+            while (
+              isWildcard(nextAppPattern) &&
+              wildcardType(nextAppPattern) === 'OptionalSequence'
+            ) {
+              if (!patterns[nextAppPatternIndex + 1]) break;
+              // @note: if pattern has been validated prior, the next should never be a '_'
+              // (universal) or '__' (regular sequence) wildcard
+              nextAppPattern = patterns[++nextAppPatternIndex];
+            }
+
+            // Set to `true` if there is at least one op. (remaining to be consumed) which matches
+            // the next pattern.
+            let found = false;
+            while (!found && j < ops.length) {
+              found =
+                matchOnce(ops[j], nextAppPattern, result, options) !== null;
+              if (!found) j += 1;
+            }
+
+            // The next pattern does not match against any of the remaining ops.
+            if (!found) {
+              // If not an optional-seq., can assume no overall match.
+              if (!isOptionalSeq) return null;
+              // Capture a 0-length match wildcard
+              //(?Never?Must indicate being at end of ops. at this point...?)
+              result = captureWildcard(argName, captureOps(0), result);
+            } else {
+              // If have encountered optional-sequences following this sequence wildcard, capture
+              // these (i.e. as corresponding to 0 operands)
+              if (nextAppPattern !== nextPattern) {
+                // Index of pattern which is an optional-sequence Wildcard:
+                let wildcardIndex = i + 1;
+                while (wildcardIndex < nextAppPatternIndex) {
+                  result = captureWildcard(
+                    wildcardName(patterns[wildcardIndex++])!,
+                    captureOps(0),
+                    result!
+                  );
+                }
+              }
+
+              // A sequence wildcard has matched up until the *first* operand that the next (valid) pattern matches.
+              // First try matching remaining ops. against remaining patterns with the sequence
+              // only capturing the operands leading up to this...
+              // Otherwise, continue to see if the next pattern matches (even) further ahead: and
+              // allow this sequence wildcard to capture even more operands.
+              // (^Necessary, for instance, if considering a pattern such as '...a + _n + b' matched
+              // against '3 + 4 + x + b'...: the sequence will have initially captured just '3', but
+              // this will result in a final overall no-match. In this case. allow the sequence to
+              // capture '3 + 4' (finally permitting a 'total' match)
+              while (j <= ops.length) {
```

Comment:
This the primary added loop permitting correct/fuller matching of sequences (outlined in this' commit message); appears sensible... ?

Reply:
Should be end of review